src/librustdoc/json/conversions.rs

```diff
@@ -41,7 +42,7 @@ impl JsonRenderer<'_> {
             })
             .collect();
         let docs = item.opt_doc_value();
-        let attrs = item.attributes(self.tcx, true);
+        let attrs = item.attributes(self.tcx, self.cache(), true);
```

Comment:
Note that my patch don't affect Rustdoc JSON. We still pass through every inert attribute. Let me know if I that needs changing.